cmd/wallet-web/src/components/Vaults/RenameVaultModal.vue

```diff
@@ -0,0 +1,114 @@
+<!--
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+-->
+
+<template>
+  <modal :show="showModal" :show-close-button="true">
+    <template #content>
+      <div
+        class="
+          flex
+          justify-between
+          items-center
+          py-4
+          px-8
+          w-full
+          border-b-2 border-neutrals-thistle
+        "
+      >
+        <span class="text-lg font-bold text-neutrals-dark">
+          {{ t('Vaults.renameVault') }}
+        </span>
+      </div>
+      <div class="flex items-center px-8 pt-10 w-full">
+        <input-field
+          v-model="vaultName"
+          :helper-message="t('Vaults.addHelperMessage')"
+          :label="t('Vaults.addlabel')"
+          :placeholder="t('Vaults.placeholderLabel')"
+          :value="vaultName"
+          type="text"
+          maxlength="42"
+          @input="updateVaultName($event)"
+        />
+      </div>
+    </template>
+    <template #actionButton>
+      <styled-button
+        class="order-first md:order-last w-full md:w-auto"
+        type="primary"
+        @click="renameVault(vaultId)"
```

Comment:
You're missing `loading` prop here. Also, can you check if the entered new name is the same as the current one and in that case skip calling the collections manager in `renameVault`?